volatility3/framework/plugins/windows/svcscan.py

```diff
@@ -89,25 +107,45 @@ def create_service_table(
         ):
             symbol_filename = "services-xp-2003-x64"
         elif (
-            versions.is_win10_16299_or_later(context=context, symbol_table=symbol_table)
+            versions.is_win10_25398_or_later(context=context, symbol_table=symbol_table)
```

Comment:
Agreed, this part felt very clunky when I was working on it. It should also be more clear in the future that the order in which the checks are executed is essential as well, I'll add a comment to that effect.

Reply:
If the order's dependent, does that mean one of the checks doesn't actually check what it says it will?  If a function is called `is_win10_25398_or_later` then it should reliably return that, or be augumented with whatever additional checks are necessary to achieve that?  Hopefully I haven't misunderstood, if you're just going through the checks find out the lowest version that passes, that's fine and obvious so everything working as intended...  5:)

Reply:
Apologies if I caused any confusion here. If my understanding is correct, I think you're correct in your interpretation - we have to fall through the checks in order from the newest OS version to the oldest. I just wanted to make it explicit that you can't reorder this list of tuples, otherwise you will likely select a version of the symbols that is too old for the sample that you're analyzing.